docs/source/reference/types.rst

```diff
@@ -100,6 +100,15 @@ the beginning or the end of the index specification::
    >>> numba.float32[::1, :, :]
    array(float32, 3d, F)
 
+Miscellaneous Types
+-------------------
```

Comment:
It looks a bit weird when rendered that there's no sentence here to say e.g. `There are some miscellaneous types that are useful but do not fit anywhere in particular in the type system.`